.github/workflows/ci.yml

```diff
@@ -22,51 +22,89 @@ jobs:
+    with:
+      go-version: ${{ matrix.go-version }}
+      platform: "ubuntu-latest"
+      project-directory: "."
+      rootless-docker: false
+      ryuk-disabled: true
 
-      - name: go test
-        # only run tests on linux, there are a number of things that won't allow the tests to run on anything else
-        # many (maybe, all?) images used can only be build on Linux, they don't have Windows in their manifest, and
-        # we can't put Windows Server in "Linux Mode" in Github actions
-        # another, host mode is only available on Linux, and we have tests around that, do we skip them?
-        if: ${{ matrix.platform == 'ubuntu-latest' }}
-        run: |
-          go install gotest.tools/gotestsum@latest
-          make test-unit
+  # The job below is a copy of the job above, but with Docker rootless.
+  # It's executed in a secondary stage to avoid concurrency issues.
+  test-rootless-docker:
+    name: "Test with Rootless Docker"
+    needs: test
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+        platform: [ubuntu-latest]
+    uses: ./.github/workflows/ci-test-go.yml
+    with:
+      go-version: ${{ matrix.go-version }}
+      platform: "ubuntu-latest"
+      project-directory: "."
+      rootless-docker: true
+      ryuk-disabled: false
 
-      - name: Run module generator tests
-        if: ${{ matrix.platform == 'ubuntu-latest' }}
-        run: make -C modulegen test-unit
+  test-module-generator:
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+        platform: [ubuntu-latest, macos-latest]
+    uses: ./.github/workflows/ci-test-go.yml
+    with:
+      go-version: ${{ matrix.go-version }}
+      platform: ${{ matrix.platform }}
+      project-directory: "modulegen"
+      rootless-docker: false
+      ryuk-disabled: false
 
-      - name: Run checker
-        run: |
-          ./scripts/check_environment.sh
+  test-modules:
+    needs: test
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+        platform: [ubuntu-latest, macos-latest]
+        module: [compose, couchbase, k3s, localstack, mysql, neo4j, postgres, pulsar, redis, redpanda, vault]
+        exclude:
+          - module: compose
+            go-version: 1.19.x
+    uses: ./.github/workflows/ci-test-go.yml
+    with:
+      go-version: ${{ matrix.go-version }}
+      platform: ${{ matrix.platform }}
+      project-directory: modules/${{ matrix.module }}
+      rootless-docker: false
+      ryuk-disabled: false
 
-      - name: Test Summary
-        uses: test-summary/action@4ee9ece4bca777a38f05c8fc578ac2007fe266f7
-        with:
-          paths: "**/TEST-unit*.xml"
-        if: always()
+  test-examples:
+    needs: test-modules
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+        platform: [ubuntu-latest, macos-latest]
+        module: [bigtable, cockroachdb, consul, datastore, firestore, mongodb, nats, nginx, pubsub, spanner, toxiproxy]
```

Comment:
I was wondering if we can auto discover the folders under examples to build the matrix automatically